arm/xarm.go

```diff
@@ -266,7 +277,17 @@ func NewXArm(ctx context.Context, name resource.Name, newConf *Config, logger lo
 		speed:        utils.DegToRad(float64(newConf.speed())),
 	}
 
-	err := x.connect(ctx)
+	if newConf.Motion != "" {
+		if deps == nil {
```

Comment:
I dont think we should fail here.  If motion is unprovided the only function that is affected is `MoveToPosition` and its easy enough to do there. Seems like this will cause friction for the usrer

Reply:
if motion is unprovided aka "" then we just ignore it.